### PR TITLE
transaction db migration

### PR DESCRIPTION
**Pull request by frankobe**
1. append `c0_d0/nedb/transactions.db` and `c0_d1/nedb/transactions.db` to `local/nedb/transactions.db`
2. rename `c0_d0/nedb/transactions.db` to `c0_d0/nedb/transactions.db.old`, `c0_d1/nedb/transactions.db` to `c0_d1/nedb/transactions.db.old` to avoid duplicated migration
3. change dev datadir to include version too.

### CODE REVIEW

File: src/db/nedb.js [hunk 1]
```diff
@@ -13,14 +14,22 @@ const db = {
 
 // Init datastores
 async function initDB() {
-  const basePath = `${Utils.getDataDir()}/nedb`;
-  logger.info(`DB path: ${basePath}`);
+  try {
+    await migrateDB();
```
Comment (dwalintukan) [inline src/db/nedb.js]: directly call `await migrateTxDB()`? also, you are awaiting a non-async func

Comment (frankhu-optimizely) [reply src/db/nedb.js]: Leave then structure here for more migration script in the future
add `async` keyword

File: src/utils/utils.js [hunk 2]
```diff
@@ -149,29 +149,45 @@ class Utils {
...
+    let basePath = `${osDataDir}/${pathPrefix}`;
+    if (this.isDevEnv()) {
+      basePath += '/dev';
+    }
+    return basePath;
+  }
+  /*
+  * Returns the path where the blockchain data directory is, and also creates the directory if it doesn't exist.
+  */
+  static getBlockChainDataDir() {
```
[the remark this answers is not in the document]
Comment (frankhu-optimizely) [reply src/utils/utils.js]: done

File: src/db/nedb.js (hunk 1 above)
Comment (frankobe) [reply src/db/nedb.js]: Leave then structure here for more migration script in the future
add async keyword

File: src/utils/utils.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (frankobe) [reply src/utils/utils.js]: done

File: src/db/nedb.js [hunk 3]
```diff
@@ -13,14 +14,22 @@ const db = {
...
+  } catch (err) {
+    throw new Error(`DB Migration Error: ${err.message}`);
+  }
+
+  const blockChainDataPath = `${Utils.getBlockChainDataDir()}/nedb`;
```
[the remark this answers is not in the document]
Comment (frankobe) [reply src/db/nedb.js]: done

**Pull request merged**